ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/WSQueryEndpoint.java

```diff
@@ -163,10 +164,23 @@ public WSQueryEndpoint(
         Objects.requireNonNull(serviceContextFactory, "serviceContextFactory");
   }
 
+  private void checkEndpointAuthorization(final Principal userPrincipal) {
+    final Class className = this.getClass();
+    final String methodName = this.getClass().getAnnotation(ServerEndpoint.class).value();
```

Comment:
Why are we using the path here when we use the method name for the http endpoints?

Reply:
Done